File: Herança - Subclasses de Personagem/Personagem.py
class Personagem:
    def __init__(self, n=None, v=0, f=0, ni=1):
        self.nome = None
        self.vida = 0
        self.forca = 0
        self.nivel = 1
        self.inventario = []

        if n is not None:
            self.setNome(n)
        self.setVida(v)
        self.setForca(f)
        self.setNiveL(ni)

    def getNome(self):
        return self.nome

    def getVida(self):
        return self.vida

    def getForca(self):
        return self.forca

    def getNivel(self):
        return self.nivel

    def getInventario(self):
        return self.inventario

    def setNome(self, n):
        if n is not None:
            self.nome = n
        else:
            print("Nome inválido!")

    def setVida(self, v):
        if v >= 0 and v <= 100:
            self.vida = v
        else:
            print("Vida inválida!")

    def setForca(self, f):
        self.forca = f

    def setNiveL(self, n):
        if n >= 1:
            self.nivel = n
        else:
            print("Nível inválida!")

    def setInventario(self, i):
        self.inventario = i

    def pegarItem(self, i):
        self.inventario.append(i)
        self.calcular_forca()

    def calcular_forca(self):
        for i in range(len(self.getInventario())):
            self.setForca(
                self.getForca() + self.getInventario()[i].getBonus()
            )

    def receberDano(self, d):
        self.vida = self.vida - d

        if self.vida < 0:
            self.vida = 0

        print(self.getNome() + " recebeu " + str(d) + " de dano")

    def atacar(self):
        self.atacar_forca(10)

    def atacar_forca(self, f):
        print(self.nome + " atacou com " + str(f) + " de força")

    def ficha(self):
        print(
            "\n====FICHA==== \nNome: " + self.getNome() +
            "\nVida: " + str(self.getVida()) +
            "\nForça: " + str(self.getForca()) +
            "\nNivel: " + str(self.getNivel()) +
            "\nInventario:"
        )

        if len(self.getInventario()) == 0:
            print("Inventario Vazio.")
        else:
            for i in range(len(self.getInventario())):
                print(self.getInventario()[0].descricao())

        print("===========")

File: Herança - Subclasses de Personagem/Personagem.java
import java.util.ArrayList;

public class Personagem{
    private String nome;
    private int vida;
    private int forca;
    private int nivel;
    private ArrayList<Item> inventario = new ArrayList<>();

    public Personagem(){
    }
    public Personagem(String n, int v, int f, int ni){
        setNome(n);
        setVida(v);
        setForca(f);
        setNiveL(ni);
    }

    public String getNome(){
        return nome;
    }
    public int getVida(){
        return vida;
    }
    public int getForca(){
        return this.forca;
    }
    public int getNivel(){
        return nivel;
    }
    public ArrayList<Item> getInventario(){
        return this.inventario;
    }
    public void setNome(String n){
        if (n!=null){
            nome = n;
        }
        else{
            System.out.println("Nome inválido!");
        }
    }
    public void setVida(int v){
        if (v>=0 && v<=100){
            vida=v;
        }
        else{
            System.out.println("Vida inválida!");
        }
    }
    public void setForca(int f){
        this.forca = f;
    }
    public void setNiveL(int n){
        if (n>=1){
            nivel=n;
        }
        else{
            System.out.println("Nível inválida!");
        }
    }
    public void setInventario(ArrayList<Item> i){
        this.inventario = i;
    }
    public void pegarItem(Item i){
        this.inventario.add(i);
        calcular_forca();
    }
    public void calcular_forca(){
        for (int i=0; i<getInventario().size(); i++){
            setForca(getForca() + getInventario().get(i).getBonus());
        }
    }

    public void receberDano(int d){
        this.vida = this.vida - d;
        if (this.vida<0){
            this.vida=0;
        }
        System.out.println(getNome() + " recebeu " + d + " de dano");
    }
    public void atacar(){
        atacar(10);
    }
    public void atacar(int f){
        System.out.println(nome + " atacou com " + f + " de força");
    }

    public void ficha(){
        System.out.println("\n====FICHA==== \nNome: " + getNome() + "\nVida: " + getVida() + "\nForça: " + getForca() + "\nNivel: " + getNivel() + "\nInventario:");
        if (getInventario().size() == 0){
            System.out.println("Inventario Vazio.");
        }
        else{
            for (int i=0; i<getInventario().size(); i++){
                System.out.println(getInventario().get(0).descricao());
            }
        }
        System.out.println("===========");
    }
}